sample-app/app/controllers/users_controller.rb

```diff
@@ -1,3 +1,27 @@
 class UsersController < ApplicationController
-  def new; end
+  def new 
+    @user = User.new
+  end
+
+  def show 
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      render "static_pages/home"
+    end
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = t "greeting" 
+      redirect_to @user
+    else
+      flash[:danger] = t "signup_fail"
+      render :new
+    end
+  end
+
+  def user_params
+    params.require(:user).permit(User::PERMIT)
```

Comment:
```suggestion
    params.require(:user).permit User::PERMIT
```
đặt const là USER_PARAMS đi e